sympy/series/tests/test_series.py

```diff
@@ -79,6 +79,27 @@ def test_issue_5223():
 
     assert exp(sin(x)*log(x)).series(n=2) == 1 + x*log(x) + O(x**2*log(x)**2)
 
+
+def test_issue_11313():
+    assert Integral(cos(x), x).series(x) == sin(x).series(x)
+    assert Derivative(sin(x), x).series(x, n=3).doit() == cos(x).series(x, n=3)
+
+    assert Derivative(x**3, x).as_leading_term(x) == 3*x**2
+    assert Derivative(x**3, y).as_leading_term(x) == 0
+    assert Derivative(sin(x), x).as_leading_term(x) == 1
+    assert Derivative(cos(x), x).as_leading_term(x) == -x
+
+    # This result is equivalent to zero, zero is not return because
+    # `Expr.series` doesn't currently detect an `x` in its `free_symbol`s.
+    assert Derivative(1, x).as_leading_term(x) == Derivative(1, x)
+
+    assert Derivative(exp(x), x).series(x).doit() == exp(x).series(x)
+    assert 1 + Integral(exp(x), x).series(x) == exp(x).series(x)
```

Comment:
Anyway, it's already like that in SymPy, this PR corrects the the length of terms of higher degree, the lowest degree of the series is unaffected.

For example, SymPy 1.0:
```python
In [1]: Integral(exp(x), x).series()
Out[1]: 
     2    3    4     5     6        
    x    x    x     x     x     ⎛ 6⎞
x + ── + ── + ── + ─── + ─── + O⎝x ⎠
    2    6    24   120   720 
```

which is weird, because the _x**6_ term should be absorbed in _O(x**6)_.

Now it's like
```python
In [5]: Integral(exp(x), x).series()
Out[5]: 
     2    3    4     5        
    x    x    x     x     ⎛ 6⎞
x + ── + ── + ── + ─── + O⎝x ⎠
    2    6    24   120  
```